### PR TITLE
Add transformations from kafka topic names to hive tables names

### PR DESCRIPTION
**Pull request by serssp**
confluentinc/kafka-connect-hdfs#155

### CODE REVIEW

**Comment by serssp** [comment]
[clabot:check]

**Comment by cricket007** [comment]
Is it not possible to do this with `RegexRouter` already?

**Comment by serssp** [comment]
> Is it not possible to do this with `RegexRouter` already?

Unfortunately not, I tried to do it, but I received the NPE just as in this task https://github.com/confluentinc/kafka-connect-hdfs/issues/155#issuecomment-386227778

**Comment by cricket007** [comment]
I see. One comment I had was the default property - should that be `[-.]`? And are there unit tests that could be added for this feature?

**Comment by serssp** [comment]
Hello, @cricket007 !

Thank you for review

I have added test.
Also I have added HiveMetaStoreException when topic name not matched. This behavior more safe and predictive.

I didn't make changes in replacement from "[.-]" to "_".  This replacement is performed after regex matching.

File: hive/src/test/java/io/confluent/connect/storage/hive/HiveMetaStoreTableNameConverterFactoryTest.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+  @Parameterized.Parameter(3)
+  public Class<Throwable> exception;
+
+  @Parameterized.Parameters(name = "pattern='{0}' sourceTableName='{1}' convertedTableName='{2}'")
+  public static Collection<Object[]> data() {
+    return Arrays.asList(new Object[][]{
+            {HIVE_TABLE_PATTERN_DEFAULT, "topic_name", "topic_name", null},
+            {HIVE_TABLE_PATTERN_DEFAULT, "topic-name", "topic_name", null},
+            {HIVE_TABLE_PATTERN_DEFAULT, "topic-name", "topic_name", null},
```
Comment (cricket007) [inline hive/src/test/java/io/confluent/connect/storage/hive/HiveMetaStoreTableNameConverterFactoryTest.java]: I assume one of these should be `topic.name`?

File: hive/src/test/java/io/confluent/connect/storage/hive/HiveMetaStoreTableNameConverterFactoryTest.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+            {"^.*\\.(.*)$", "topic.name", "name", null},
+            {"^.*\\.(.*)$", "topic-name", null, HiveMetaStoreException.class}
+    });
+  }
+
+  @Test
+  public void test() {
+    HiveConfig hiveConfig = new HiveConfig(new HashMap<String, String>() {{
+      put(HIVE_TABLE_PATTERN_CONFIG, hivePatternConfig);
+    }});
```
Comment (cricket007) [inline hive/src/test/java/io/confluent/connect/storage/hive/HiveMetaStoreTableNameConverterFactoryTest.java]: Double bracket initialization is considered an anti-pattern in some contexts. I would say just make the HashMap and call `.put`

File: hive/src/test/java/io/confluent/connect/storage/hive/HiveMetaStoreTableNameConverterFactoryTest.java (hunk 1 above)
Comment (serssp) [reply hive/src/test/java/io/confluent/connect/storage/hive/HiveMetaStoreTableNameConverterFactoryTest.java]: Yes, thx!

**Pull request closed without merging**